Client/game_sa/CFxSA.cpp

```diff
@@ -239,3 +239,93 @@ void CFxSA::TriggerFootSplash(CVector& vecPosition)
+{
+    // Init our own FxPrtMult struct
+    FxPrtMult_c fxPrt{{fR,fG,fB,fA}, fSize, 0, fLife};
+    CVector     newDirection;
+
+    FxSystem_c* fxParticleSystem;
+
+    switch (eFxParticle)
+    {
+        case PRT_BLOOD:
+            fxParticleSystem = m_pInterface->m_fxSysBlood;
+            break;
+        case PRT_BOATSPLASH:
+            fxParticleSystem = m_pInterface->m_fxSysBoatSplash;
+            break;
+        case PRT_BUBBLE:
+            fxParticleSystem = m_pInterface->m_fxSysBubble;
+            break;
+        case PRT_DEBRIS:
+            fxParticleSystem = m_pInterface->m_fxSysDebris;
+            break;
+        case PRT_GUNSHELL:
+            fxParticleSystem = m_pInterface->m_fxSysGunshell;
+            break;
+        case PRT_SAND:
+            fxParticleSystem = m_pInterface->m_fxSysSand;
+            break;
+        case PRT_SAND2:
+            fxParticleSystem = m_pInterface->m_fxSysSand2;
+            break;
+        case PRT_SMOKE:
+            fxParticleSystem = m_pInterface->m_fxSysSmoke;
+            break;
+        case PRT_SMOKEHUGE:
+            fxParticleSystem = m_pInterface->m_fxSysSmokeHuge;
+            break;
+        case PRT_SMOKE2:
+            fxParticleSystem = m_pInterface->m_fxSysSmoke2;
+            break;
+        case PRT_SPARK:
+            fxParticleSystem = m_pInterface->m_fxSysSpark;
+            break;
+        case PRT_SPARK2:
+            fxParticleSystem = m_pInterface->m_fxSysSpark2;
+            break;
+        case PRT_SPLASH:
+            fxParticleSystem = m_pInterface->m_fxSysSplash;
+            break;
+        case PRT_WAKE:
+            fxParticleSystem = m_pInterface->m_fxSysWake;
+            break;
+        case PRT_WATERSPLASH:
+            fxParticleSystem = m_pInterface->m_fxSysWaterSplash;
+            break;
+        case PRT_WHEELDIRT:
+            fxParticleSystem = m_pInterface->m_fxSysWheelDirt;
+            break;
+        case PRT_GLASS:
+            fxParticleSystem = m_pInterface->m_fxSysGlass;
+            break;
+        default:
+            fxParticleSystem = m_pInterface->m_fxSysBlood;
+    }
+
+    for (int i = 0; i < iCount; i++)
+    {
+        if (bRandomizeColors)
+        {
+            // 0x49EECB
+            fxPrt.m_color.red = (rand() % 10000) * 0.0001 * fR + 0.13;
+            fxPrt.m_color.green = (rand() % 10000) * 0.0001 * fG + 0.12;
+            fxPrt.m_color.blue = (rand() % 10000) * 0.0001 * fB + 0.04;
+        }
+
+        if (bRandomizeSizes)
+            // 0x49EF21 - Calculate random size for each particle
+            fxPrt.m_fSize = (rand() % 10000) * 0.0001 * fSize + 0.3;
+
+        // 0x49EF4C - Calculate random direction for each particle
+        newDirection = CVector(vecDirection.fX * 4, vecDirection.fY * 4, vecDirection.fZ * 4);
```

Comment:
Where did you find this algorithm? What does the multiplication by 4 means?

Reply:
Okay. I see that this logic is inherited from SA.

Reply:
I copied it from the original SA code, e.g. AddWood (0x49EE10).
![image](https://github.com/multitheftauto/mtasa-blue/assets/26044224/c35204f9-fa69-480b-ba57-e43337bcf5eb)